pymc3/tests/test_memo.py

```diff
@@ -11,21 +11,59 @@
 #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
+import numpy as np
+import pymc3 as pm
 
-from pymc3.memoize import memoize
+from pymc3 import memoize
 
 
-def getmemo():
-    @memoize
-    def f(a, b=("a")):
-        return str(a) + str(b)
+def test_memo():
+    def fun(inputs, suffix="_a"):
+        return str(inputs) + str(suffix)
 
-    return f
+    inputs = ["i1", "i2"]
+    assert fun(inputs) == "['i1', 'i2']_a"
+    assert fun(inputs, "_b") == "['i1', 'i2']_b"
 
+    funmem = memoize.memoize(fun)
+    assert hasattr(fun, "cache")
+    assert isinstance(fun.cache, dict)
+    assert len(fun.cache) == 0
+
+    # call the memoized function with a list input
+    # and check the size of the cache!
+    assert funmem(inputs) == "['i1', 'i2']_a"
+    assert funmem(inputs) == "['i1', 'i2']_a"
+    assert len(fun.cache) == 1
+    assert funmem(inputs, "_b") == "['i1', 'i2']_b"
+    assert funmem(inputs, "_b") == "['i1', 'i2']_b"
+    assert len(fun.cache) == 2
+
+    # add items to the inputs list (the list instance remains identical !!)
+    inputs.append("i3")
+    assert funmem(inputs) == "['i1', 'i2', 'i3']_a"
+    assert funmem(inputs) == "['i1', 'i2', 'i3']_a"
+    assert len(fun.cache) == 3
 
-def test_memo():
-    f = getmemo()
 
-    assert f("x", ["y", "z"]) == "x['y', 'z']"
-    assert f("x", ["a", "z"]) == "x['a', 'z']"
-    assert f("x", ["y", "z"]) == "x['y', 'z']"
+def test_hashing_of_rv_tuples():
+    obs = np.random.normal(-1, 0.1, size=10)
+    with pm.Model() as pmodel:
+        mu = pm.Normal("mu", 0, 1)
+        sd = pm.Gamma("sd", 1, 2)
+        dd = pm.DensityDist(
+            "dd",
+            pm.Normal.dist(mu, sd).logp,
+            random=pm.Normal.dist(mu, sd).random,
+            observed=obs,
+        )
+        print()
```

Comment:
do these print statements have any effect in this test?

Reply:
I added this print because the first print ended up on the same line when running `pytest -v -s`. It was hard to read..

Reply:
And I printed because otherwise one doesn't know which variable caused the stackoverflow